espaloma/graphs/deploy.py

```diff
@@ -0,0 +1,171 @@
+# =============================================================================
+# IMPORTS
+# =============================================================================
+import rdkit
+import torch
+from openforcefield.typing.engines.smirnoff import ForceField
+import espaloma as esp
+from simtk import unit
+from simtk.unit.quantity import Quantity
+
+# =============================================================================
+# CONSTANTS
+# =============================================================================
+OPENMM_LENGTH_UNIT = unit.nanometer
+OPENMM_ANGLE_UNIT = unit.radian
+OPENMM_ENERGY_UNIT = unit.kilojoule_per_mole
+
+OPENMM_BOND_EQ_UNIT = OPENMM_LENGTH_UNIT
+OPENMM_ANGLE_EQ_UNIT = OPENMM_ANGLE_UNIT
+OPENMM_TORSION_K_UNIT = OPENMM_ENERGY_UNIT
+OPENMM_TORSION_PHASE_UNIT = OPENMM_ANGLE_UNIT
+OPENMM_BOND_K_UNIT = OPENMM_ENERGY_UNIT / (OPENMM_LENGTH_UNIT ** 2)
+OPENMM_ANGLE_K_UNIT = OPENMM_ENERGY_UNIT / (OPENMM_ANGLE_UNIT ** 2)
+
+# =============================================================================
+# MODULE FUNCTIONS
+# =============================================================================
+def openmm_system_from_graph(g, forcefield="openff_unconstrained-1.1.0", suffix=""):
+    """ Construct an openmm system from `espaloma.Graph`.
+
+    Parameters
+    ----------
+    g : `espaloma.Graph`
+        Input graph.
+
+    Returns
+    -------
+    sys : `openmm.System`
+        Constructed single-molecule OpenMM system.
+
+
+    """
+    # get a forcefield
+    try:
+        ff = ForceField("%s.offxml" % forcefield)
+    except:
+        try:
+            ff = ForceField("test_forcefields/%s.offxml" % forcefield)
+        except:
+            raise NotImplementedError
```

Comment:
should print or log or save which forcefield was used, especially if it falls back from the one specified

Reply:
Actually this is only because smirnoff99frosst and openff don't live in the same place. There is no falling back logic being implemented here.

Reply:
Got it -- thanks!